Chat/BroadcastSocket.cpp

```diff
@@ -0,0 +1,63 @@
+#include "BroadcastSocket.h"
+
+BroadcastSocket::BroadcastSocket(int port_to_send)
+{
+    memset(&broadcast_addr_, 0, sizeof(broadcast_addr_));
+    broadcast_addr_.sin_family = AF_INET;   //ipv4
+    broadcast_addr_.sin_port = 
+        htons(port_to_send != -1 ? (unsigned int)port_to_send : PORT); //defined in Defines.h
+    
+    //wanted to change it to inet_pton, but it doesn't work for some reason
+    //BROADCAST_IP defined in Defines.h
+    broadcast_addr_.sin_addr.s_addr = inet_addr(BROADCAST_IP);
+}
+
+BroadcastSocket::~BroadcastSocket()
+{}
+
+int BroadcastSocket::Initialize()
```

Comment:
If it doesn't matter if it will be -1 or false, please make this function boolean too.